docs/DevGuide/TagsTypelists.md

```diff
@@ -0,0 +1,190 @@
+that does not exist! As written however, the program is well-formed and there
+is no error emitted at compilation. Because the keys are indistinguishable
+from their type alone, the mistake is only caught at runtime. The runtime
+portion of a SpECTRE calculation will typically be much longer (up to thousands
+of times longer!) than the compile time portion, so it is critical to catch
+costly mistakes like this as early into the calculation as possible.
+SpECTRE takes advantage of C++'s static typing of to help users distinguish
+between quantities at the type level. This is done through tags.
+
+An example of a tag looks like the following:
+
+~~~{.c}
+struct Pressure {
+  //The type of the quantity to which the tag refers:
+  using type = double;
+};
+
+~~~
+
+Note that through the use of tags, we have now promoted `Pressure` from being
+a *value* used to initialize a `std::string` at runtime, to a *type*
+distinguishable from other types at compile time. A large portion of SpECTRE is
+designed with the philosophy of enlisting the help of the compiler in assuring
+the correctness of our programs.
+
+# Tags in SpECTRE {#documentation_for_tags_in_spectre}
+
+## OptionTags {#documentation_for_option_tags}
+You'll find an example of option tags used in the SingletonHelloWorld example
+executable. For convenience, we provide it here:
+
+\snippet SingletonHelloWorld.hpp executable_example_options
+
+Option tags reflect options that are specifiable at runtime through YAML input
+files. Option tags are used to specify runtime settings for the Domain,
+AnalyticSolution, and EventsAndTriggers, among other things.
+
+## InboxTags {#documentation_for_inbox_tags}
+SpECTRE is designed to run on multiple processors in parallel. By default then,
+SpECTRE assumes that data created on one processor is not needed to be
+communicated to any other processor. However, some programs need data to be
+communicated between processors, such as flux data. An inbox tag follows the
+general pattern:
+
+~~~{.c}
+struct TypeRecieveTag {
+  using temporal_id = int;
+  using type = std::unordered_map<temporal_id, TypeToReceive>;
+};
+
+~~~
+
+## SimpleTags {#documentation_for_simple_tags}
+The `naive_databox` shown above was a container useful for retrieving
+quantities using the names of the quantities as keys. SpECTRE's DataBox
+uses SimpleTags, which are *types*, as keys, whereas the `naive_databox`'s keys
+were simply *values* of type `std::string`. Here is an example of a SimpleTag:
+
+\snippet Test_DataBox.cpp databox_tag_example
+
+## ComputeTags {#documentation_for_compute_tags}
+The previous tags mentioned are used to tag quantities which are obtained
+through specification via options (OptionTags), through being received
+from another processor (InboxTags), or through retrieval from a DataBox
+(SimpleTags). Quantities can also be produced through computations, and
+ComputeTags tag the functions used to compute them. A compute tag can look
```

Comment:
I think this list of tag types would be useful at the beginning of the section but not in the `ComputeTags` subsection.